graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```
@@ -147,55 +147,64 @@ class InputTypeGenerator(config: CodeGenConfig, document: Document) : BaseDataTy
+                                        "Property \"${objectProperty.name}\" does not exist in input type \"${inputObjectDefinition.name}\""
+                                    }
+                                val argumentValue = generateCode(
+                                    objectProperty.value,
+                                    typeUtils.findReturnType(argumentType.type),
+                                    inputTypeDefinitions
+                                )
+                                "set${objectProperty.name.replaceFirstChar { it.uppercaseChar() }}($argumentValue);"
+                            } + "}}"
+                        )
+                    }
                 }
+                else -> CodeBlock.of("\$L", value)
             }
-            else -> CodeBlock.of("\$L", value)
-        }
     }
 
-    private fun checkAndGetLocaleValue(value: StringValue, type: JavaTypeName): String? {
-        if (type.toString() == "java.util.Locale") return "Locale.forLanguageTag(\"${value.value}\")"
-        return null
+    private fun checkAndGetLocaleCodeBlock(value: Value<out Value<*>>, type: JavaTypeName): CodeBlock? {
+        return if (value is StringValue && type.toString() == "java.util.Locale") {
+            CodeBlock.of("\$L", "Locale.forLanguageTag(\"${value.value}\")")
+        } else null
+    }
+
+    private fun checkAndGetBigDecimalCodeBlock(value: Value<out Value<*>>, type: JavaTypeName): CodeBlock? {
+        return if (type.toString() == "java.math.BigDecimal") {
+            when (value) {
+                is StringValue -> CodeBlock.of("new java.math.BigDecimal(\$S)", value.value)
+                is IntValue -> CodeBlock.of("new java.math.BigDecimal(\$L)", value.value)
+                is FloatValue -> CodeBlock.of("new java.math.BigDecimal(\$L)", value.value)
+                else -> null
```

Same thought here around throwing an exception rather than silently defaulting to null.